storage_json.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	model "github.com/golang-bristol/beer-model"
+	"github.com/nanobox-io/golang-scribble"
+)
+
+const (
+	// CollectionBeer identifier for JSON collection about beers
+	CollectionBeer int = iota
+	// CollectionReview identifier for JSON collection about reviews
+	CollectionReview
+)
+
+// StorageJSON is the data storage type using JSON file
+type StorageJSON struct {
+	db *scribble.Driver
+}
+
+func newStorageJSON(location string) (*StorageJSON, error) {
+	var err error
+
+	stg := new(StorageJSON)
+
+	stg.db, err = scribble.New(location, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return stg, nil
+}
+
+// SaveBeer insert new beers
+func (s *StorageJSON) SaveBeer(beers ...model.Beer) error {
+	for _, beer := range beers {
+		var resource = strconv.Itoa(beer.ID)
+		var collection = strconv.Itoa(CollectionBeer)
+
+		allBeers := s.FindBeers()
+		for _, b := range allBeers {
+			if beer.Abv == b.Abv &&
+				beer.Brewery == b.Brewery &&
+				beer.Name == b.Name {
+				return fmt.Errorf("Beer already exists")
+			}
+		}
+
+		// TODO: Since delete function has not been implemented yet
+		// I think we can assume size of beers should always increase.
+		beer.ID = len(allBeers) + 1
```

Comment:
This is ok to assume for now. A better way would be to make the cellar a map of ID -> beer, so we don't have to worry about ID clashes. But that's something for the future :)

Reply:
(Also I only thought of this now ha :D )